tests/test_git/common.py

```diff
@@ -0,0 +1,60 @@
+import sublime
+import subprocess
+import os
+import shutil
+import tempfile
+from unittesting import DeferrableTestCase
+
+
+# copy from https://github.com/randy3k/ProjectManager/blob/f8667697a77a9e7020891e3227de25c2ebb990b3/pm.py#L72
+def subl(args=[]):
+    executable_path = sublime.executable_path()
+    if sublime.platform() == 'osx':
+        app_path = executable_path[:executable_path.rfind(".app/") + 5]
+        executable_path = app_path + "Contents/SharedSupport/bin/subl"
+    subprocess.Popen([executable_path] + list(args))
+    if sublime.platform() == "windows":
+        def fix_focus():
+            window = sublime.active_window()
+            view = window.active_view()
+            window.run_command('focus_neighboring_group')
+            window.focus_view(view)
+        sublime.set_timeout(fix_focus, 300)
+
+
+class GSCreateRepoTestCase(DeferrableTestCase):
+
+    _temp_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup a repo for testing
+        """
+        cls._temp_dir = tempfile.mkdtemp()
+        subl(["-n", cls._temp_dir])
+
+        def condition():
+            for d in sublime.active_window().folders():
+                # on Windows, `cls._temp_dir` is lowered cased,
+                # `os.path.normcase` is needed for comparison.
+                if cls._temp_dir == os.path.normcase(d):
+                    return True
+
+        yield condition
```

Comment:
Could you kindly explain what is going on here? It looks like we're spawning a new sublime window and then iterating over windows to find the window with the path we gave it as an argument, but I may be wrong.

Why are we yielding a callable in the `setUpClass`? Perhaps this can be separate method with a clear functionality that can be called when setting up the testcase class?